core/rust/qdbr/src/parquet_read/meta.rs

```diff
@@ -51,8 +60,27 @@ impl ParquetDecoder {
         Ok(decoder)
     }
 
-    fn descriptor_to_column_type(des: &Descriptor) -> Option<ColumnType> {
-        match (
+    fn descriptor_to_column_type(
+        des: &Descriptor,
+        additional_meta: &Option<HashMap<String, Option<String>>>,
+    ) -> Option<(ColumnType, i32)> {
+        let result = additional_meta.as_ref().and_then(|hashmap| {
+            hashmap.get(&(QDB_TYPE_META_PREFIX.to_string() + &des.primitive_type.field_info.name))
+        });
+        if let Some(Some(val)) = result {
+            let col_type = match val.parse::<i32>() {
+                Ok(col_type_int) => match ColumnType::try_from(col_type_int) {
+                    Ok(col_type) => return Some((col_type, col_type_int)),
```

Comment:
type unsafe

Reply:
Not sure I see type unsafeness here, can you explain?

Reply:
We talked about it on the phone, I believe we agreed that it's not unsafe but can be potentially restrictive in some future use and we will improve it then.

Reply:
the `val` value can be not a valid column type encoding.
ok to keep it as it is now.